### ISSUE TITLE
ISplitFactoryProps.children type is overly restrictive

### ISSUE THREAD

**Issue by jamesarosen** [splitio/react-client#164]
The type is [defined](https://github.com/splitio/react-client/blob/bb83870456716ec560d5edabf29c56c8d3401e58/types/types.d.ts#L103-L106) as

```ts
/**
 * Children of the SplitFactory component. It can be a functional component (child as a function) or a React element.
 */
children: ((props: ISplitFactoryChildProps) => JSX.Element | null) | JSX.Element | null;
```

That doesn't include `ReactNode`, which makes it incompatible with, e.g. `PropsWithChildren` and `FC`:

```ts
// React types:
type ReactNode = ReactChild | ReactFragment | ReactPortal | boolean | null | undefined;
type PropsWithChildren<P> = P & { children?: ReactNode };
```

This won't pass type-checking:

```ts
const MySplitContext: FC = ({ children }) => {
  const user = useContext(UserContext);

  const splitConfig: SplitIO.IBrowserSettings = {
    core: {
      authorizationKey: process.env.SPLIT_AUTHORIZATION_KEY,
      key: user.email,
    }
  }

  return (
    <SplitFactory config={splitConfig} attributes={user}>
      {children}
    </SplitFactory>
  )
}
```

<img width="1095" alt="Screenshot 2023-10-10 at 14 30 02" src="https://github.com/splitio/react-client/assets/2406/52b27efb-7144-49dc-ac03-3171929e1f41">

One workaround is to wrap `children` (a `ReactNode`, per `FC` and `PropsWithChildren`) in a `Fragment`:

```ts
return (
  <SplitFactory config={splitConfig} attributes={user}>
    <>{children}</>
  </SplitFactory>
)
```

The customary practice is to include `ReactNode` in the list of possible children unless you _absolutely_ need to restrict the possible children, e.g. because you're copying or manipulating them. In this case, I suspect the following would work well:

```ts
children: ((props: ISplitFactoryChildProps) => ReactNode) | ReactNode
```

**Comment by EmilianoSanchez** [comment]
Hi @jamesarosen

Thank you for your suggestion. We will include it in our next release. PR [here](https://github.com/splitio/react-client/pull/165), but there is not ETA yet for the release.

**Comment by EmilianoSanchez** [comment]
Hi @jamesarosen 
FYI, we have released React SDK v1.10.0, which includes your suggestion. I will now close the issue.

Thank you
